packages/google_sign_in/tizen/src/secure_storage.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "secure_storage.h"
+
+#include <ckmc/ckmc-manager.h>
+#include <ckmc/ckmc-type.h>
+
+#include <cassert>
+
+namespace {
+
+// This value must match `_kIvSizeBytes` in `secure_storage.dart`.
+const size_t kIvSizeBytes = 12;
+
+const char *kAesKey = "AesKey";
+
+const ckmc_policy_s kDefaultKeyPolicy = {
+    .password = nullptr,
+    .extractable = false,
+};
+
+const ckmc_policy_s kDefaultDataPolicy = {
+    .password = nullptr,
+    .extractable = true,
+};
+
+enum class NameType {
+  kKey,
+  kData,
+};
+
+std::vector<std::string> GetNames(NameType name_type) {
+  ckmc_alias_list_s *ckmc_alias_list;
+  switch (name_type) {
+    case NameType::kKey:
+      ckmc_get_key_alias_list(&ckmc_alias_list);
+      break;
+    case NameType::kData:
+      ckmc_get_data_alias_list(&ckmc_alias_list);
+      break;
+    default:
+      assert(false);
+  }
+
+  std::vector<std::string> names;
+  ckmc_alias_list_s *current = ckmc_alias_list;
+  while (current != nullptr) {
+    std::string name = current->alias;
+    names.push_back(name.substr(name.find(' ') + 1));
+    current = current->next;
+  }
+
+  ckmc_alias_list_all_free(ckmc_alias_list);
+  return names;
+}
+}  // namespace
+
+SecureStorage::SecureStorage() {
+  ckmc_generate_new_params(CKMC_ALGO_AES_GCM, &params_);
+}
+
+SecureStorage::~SecureStorage() { ckmc_param_list_free(params_); }
+
+void SecureStorage::CreateKey() {
+  ckmc_create_key_aes(256, kAesKey, kDefaultKeyPolicy);
+}
+
+bool SecureStorage::HasKey() const {
+  std::vector<std::string> names = GetNames(NameType::kKey);
+  for (const auto &name : names) {
+    if (name == kAesKey) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void SecureStorage::Destroy() {
+  std::vector<std::string> names = GetDataNames();
+  for (const auto &name : names) {
+    RemoveData(name);
+  }
+  ckmc_remove_alias(kAesKey);
+}
+
+void SecureStorage::SaveData(const std::string &name, std::vector<uint8_t> data,
+                             std::vector<uint8_t> initialization_vector) {
+  std::vector<uint8_t> encrypted = EncryptData(data, initialization_vector);
+
+  ckmc_raw_buffer_s *buffer;
+  ckmc_buffer_new(encrypted.data(), encrypted.size(), &buffer);
+  int ret = ckmc_save_data(name.c_str(), *buffer, kDefaultDataPolicy);
+  if (ret == CKMC_ERROR_DB_ALIAS_EXISTS) {
+    RemoveData(name.c_str());
+    ckmc_save_data(name.c_str(), *buffer, kDefaultDataPolicy);
+  }
+  ckmc_buffer_free(buffer);
+}
+
+std::optional<std::vector<uint8_t>> SecureStorage::GetData(
+    const std::string &name) {
+  ckmc_raw_buffer_s *buffer;
```

Comment:
Could you please initialize local pointer variable with `nullptr`?

Reply:
Fixed.